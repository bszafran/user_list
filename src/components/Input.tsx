import React from "react"
import styled from "styled-components/macro"

interface InputProps {
  value: string
  onChange: (value: string) => void
  placeholder?: string
}

const InputContainer = styled.input`
  border: 1px solid black;
  font-size: 15px;
  padding: 2px 4px;

  ::placeholder {
    color: #b8b8b8;
  }
`

const Input = (props: InputProps) => {
  const { value, placeholder } = props

  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const newValue = event.target.value
    props.onChange(newValue)
  }

  return (
    <InputContainer
      value={value}
      onChange={onChange}
      placeholder={placeholder}
    />
  )
}

export default Input
