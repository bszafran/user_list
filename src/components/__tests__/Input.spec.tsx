import React from "react"
import { mount } from "enzyme"
import Input from "../Input"

describe("Input component", () => {
  const mockedInputProps = {
    value: "hello",
    onChange: () => {},
    placeholder: "Search me",
  }

  it("renders correctly with mocked props", () => {
    const inputComponent = mount(<Input {...mockedInputProps} />)
    expect(inputComponent).toMatchSnapshot()
  })

  it("fires onChange method onClick", () => {
    const onChangeSpy = jest.fn((value) => {})
    const valueToFill = "testValue"
    const inputComponent = mount(
      <Input {...mockedInputProps} onChange={onChangeSpy} />
    )

    inputComponent.find("input").simulate("change", {
      target: { value: valueToFill },
    })

    expect(onChangeSpy).toHaveBeenCalledWith(valueToFill)
  })

  it("fires onChange method with empty string when text cleared", () => {
    const onChangeSpy = jest.fn((value) => {})
    const valueToFill = "testValue"
    const inputComponent = mount(
      <Input {...mockedInputProps} onChange={onChangeSpy} />
    )

    inputComponent.find("input").simulate("change", {
      target: { value: valueToFill },
    })
    expect(onChangeSpy).toHaveBeenCalledWith(valueToFill)

    inputComponent.find("input").simulate("change", {
      target: { value: "" },
    })
    expect(onChangeSpy).toHaveBeenCalledWith("")
  })
})
