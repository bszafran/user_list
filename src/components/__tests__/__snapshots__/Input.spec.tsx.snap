// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders correctly with mocked props 1`] = `
<Input
  onChange={[Function]}
  placeholder="Search me"
  value="hello"
>
  <Input__InputContainer
    onChange={[Function]}
    placeholder="Search me"
    value="hello"
  >
    <input
      className="Input__InputContainer-r8mf24-0 gopIUt"
      onChange={[Function]}
      placeholder="Search me"
      value="hello"
    />
  </Input__InputContainer>
</Input>
`;
